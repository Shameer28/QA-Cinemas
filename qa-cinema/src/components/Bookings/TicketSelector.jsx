import { useEffect, useState } from "react";
import { Button, Container } from "react-bootstrap";
import movieUtils from "./../../utils/movies";
import payment from "../../utils/payment";
import Carousel from 'react-bootstrap/Carousel'

import image1 from './img/theatre.jpg'
import image2 from './img/set.png'
import image3 from './img/cinemas.jpg'
import image4 from './img/cinema2.jpg'
import Card from 'react-bootstrap/Card'

const TicketSelector = (props) => {

    // Get All .select only showingTimes & title
    // Axios
    // const data = [{"id":-1, "Title":"Detective", "ShowTimes":[81452812842129,20]},{"id":2, "Title":"Picka", "ShowTimes":[81452812842129]}];
    const [movies, setMovies] = useState([]);
    const [showtimes, setShowtiemes] = useState(<p>Loading Show Times</p>)
    const { setPage, getCart, setCart, url } = props;

    const getData = () => {
        setMovies([]);
        movieUtils.getAll().then((res) => {
            setMovies(res.data)
        }).catch((err) => {
            console.log(err)
        });
    }

    useEffect(() => {
        getData();
    }, [])

    const updateForm = (e) => {
        e.preventDefault();
        const firstOption = e.target.parentElement.querySelector("#MovieId")[0]
        if (firstOption.value === "Select A Movie") {
            firstOption.remove()
        }
        movieUtils.get(e.target.value).then((res) => {
            if (res.data) {
                const timming = e.target.parentElement.querySelector("#MovieTimings");
                timming.innerHTML = "";
                res.data.showTimes.map((t) => {
                    const time = new Date(t*1000)
                    timming.innerHTML += `<option>${time.toLocaleDateString()+" - "+time.getHours()+":"+time.getMinutes()}</option>`
                })
            } else {
                throw new Error("No body on responce")
            }
        }).catch((err) => {
            console.log(err)
        });

    }

    const handleSubmit = (e) => {
        e.preventDefault();

        let formData = {};

        for (let i = 0; i < 6; i++) {
            if (e.target[i].value == "Select A Movie" || e.target[i].value == "" || e.target[i].value < 0) {
                alert("Please Fill In All Inputs");
                return;
            }
            formData[e.target[i].id] = e.target[i].value;
            if (i > 1) {
                e.target[i].value = "";
            } else if (i === 0) {
                formData["MovieTitle"] = e.target[0][e.target[0].selectedIndex].text
            }

        }
        formData.price = (formData.Adults * 10) + (formData.Child * 5) + [0, 3, 5, 7.5][formData.Concessions]
        setCart(getCart => [...getCart, formData])
    }

    return (
        <div class="background">
            <Carousel controls={false} slide={true} fade={true} pause={false} height="650px">
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={image1}
                        alt="First slide"
                    />
                </Carousel.Item>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={image2}
                        alt="Second slide"
                    />
                </Carousel.Item>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={image3}
                        alt="Third slide"
                    />
                </Carousel.Item>
                <Carousel.Item interval={3000}>
                    <img
                        className="d-block w-100"
                        src={image4}
                        alt="Third slide"
                    />
                </Carousel.Item>

            </Carousel>

            <br></br>
            <h2 style={{ color: "white", fontWeight: "400", fontfamily: "isonormregular, sans-serif", letterSpacing: ".15em", textTransform: "uppercase", lineHeight: "1.1" }}>Book your tickets </h2>
            <br></br>

            <Container>
                <Card className="text-center">
                    <Card.Header>Ticket Select</Card.Header>
                    <Card.Body>
                        <Card.Title><u>Please fill out the form to book tickets</u></Card.Title>
                        <Card.Text>

                            {movies.length === 0 ? (
                                <h5>Loading Form</h5>
                            ) : (

                                <form onSubmit={handleSubmit}>

                                    <select id="MovieId" onChange={updateForm}>
                                        <option value={null}> Select A Movie </option>
                                        {movies.map((m, key) => {
                                            return <option key={key} value={String(m._id)}> {m.title} </option>
                                        })}
                                    </select>
                                    <select id="MovieTimings">
                                    </select><br />

                                    <label htmlFor="Name" >Name:</label><br />
                                    <input type="text" id="Name" /><br />

                                    <label htmlFor="Adults" >Adults £10:</label><br />
                                    <input type="number" id="Adults" /><br />

                                    <label htmlFor="Child" >Child (Under 16s) £5:</label><br />
                                    <input type="number" id="Child" /><br />

                                    <label htmlFor="Concessions" >Concession:</label><br />
                                    <select id="Concessions">
                                        <option value={0}> No Snacks </option>
                                        <option value={1}>£3 Large Drink </option>
                                        <option value={2}>£5 Popcorn </option>
                                        <option value={3}>£7.50 Large Drink & Popcorn </option>
                                    </select><br />

                                    <br /><input type="submit" value="Add To Cart" />
                                    <button onClick={() => { setPage("Checkout") }} disabled={getCart.length === 0}>Checkout</button>
                                </form>

                            )}
                        </Card.Text>
                    </Card.Body>
                    <Card.Footer className="text-muted"></Card.Footer>
                </Card>
                {/* import the andrews form if it messes up */}
            </Container>
        </div>
    );

}

export default TicketSelector;