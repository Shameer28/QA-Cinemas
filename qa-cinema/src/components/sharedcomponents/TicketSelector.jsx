import { useEffect, useState } from "react";
import axios from "axios";


const TicketSelector = (props) => {

    // Get All .select only showingTimes & title
    // Axios
    const data = [{"id":-1, "Title":"Detective", "ShowTimes":[81452812842129,20]},{"id":2, "Title":"Picka", "ShowTimes":[81452812842129]}];
    const [movies, setMovies] = useState([]);
    const {getCart, setCart, url} = props;

    const getData = () => {
        setMovies([]);
		axios.get(url).then((res) => {
            setMovies(res.body);
		}).catch((err) => {
			console.log(err)
		});
	}

    useEffect(()=>{
        getData();
    },[])


    let slots = [];
    data.map(movie => {
        movie.ShowTimes.forEach(time => {
            slots.push([movie.Title, time])
        })
    })

    
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log("submited");
    }

    // Alternative - Add On change to select
    //             - Make value id of mvoie selected
    //             - On change => updates the timing dropdown


    return (      
    <div>
    <h3>Ticket Select</h3>
    {movies.length != 0 ? (
        <h5>Loading Form</h5>
      ) : (
        <form onSubmit={handleSubmit}>

            <select id="MovieTitle">
            {slots.map(s => {
                return <option value={s}> {`${s[0]} - ${String(s[1])}`} </option>
            })}
            </select><br/>

            <label htmlFor="Name" >Name:</label><br/>
            <input type="text" id="Name" /><br/>

            <label htmlFor="Adults" >Adults:</label><br/>
            <input type="number" id="Adults" /><br/>

            <label htmlFor="Child" >Child:</label><br/>
            <input type="number" id="Child" /><br/>

            {/* Drop Down For Concessions */}

            <input type="submit" value="Submit" />
        </form>
      )}
    </div>);
}
 
export default TicketSelector;