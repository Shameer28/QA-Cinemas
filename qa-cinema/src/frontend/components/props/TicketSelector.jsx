const TicketSelector = (props) => {

    // Get All Data () .select only showingTimes
    // Axios
    const data = [{"id":-1, "Title":"Detective", "ShowTimes":[81452812842129,20]},{"id":2, "Title":"Picka", "ShowTimes":[81452812842129]}];

    let slots = [];
    data.map(movie => {
        movie.ShowTimes.forEach(time => {
            slots.push([movie.Title, time])
        })
    })

    console.log(slots);

    const {getCart, setCart} = props;

    let timings = (<div></div>)
    
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log("submited");
        console.log(e);
        console.log(getCart)
    }


    

    return (      
    <div>
    <h3>Ticket Select</h3>
    <form onSubmit={handleSubmit}>

        <select id="MovieTitle">
        {slots.map(s => {
            return <option value={s}> {`${s[0]} - ${String(s[1])}`} </option>
        })}
        </select><br/>

        <label htmlFor="Name" >Name:</label><br/>
        <input type="text" id="Name" /><br/>

        <label htmlFor="Adults" >Adults:</label><br/>
        <input type="number" id="Adults" /><br/>

        <label htmlFor="Child" >Child:</label><br/>
        <input type="number" id="Child" /><br/>

        {/* Drop Down For Concessions */}

        <input type="submit" value="Submit" />
      </form>
    </div>);
}
 
export default TicketSelector;