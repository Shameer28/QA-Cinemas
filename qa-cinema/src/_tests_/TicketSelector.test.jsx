import TicketSelector from '../components/Bookings/TicketSelector'
import { create } from 'react-test-renderer';


describe(`Ticket Selector testing`, () => {

    it(`Should match the snapshot`, () => {
        const instance = create(<TicketSelector/>).toJSON();
        expect(instance).toMatchSnapshot();
    })
})