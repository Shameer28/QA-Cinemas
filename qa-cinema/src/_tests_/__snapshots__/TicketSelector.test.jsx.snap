// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket Selector testing Should match the snapshot 1`] = `
<div
  class="background"
>
  <div
    className="carousel slide carousel-fade"
    height="650px"
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="active carousel-item"
        interval={3000}
      >
        <img
          alt="First slide"
          className="d-block w-100"
          src="theatre.jpg"
        />
      </div>
      <div
        className="carousel-item"
        interval={3000}
      >
        <img
          alt="Second slide"
          className="d-block w-100"
          src="set.png"
        />
      </div>
      <div
        className="carousel-item"
        interval={3000}
      >
        <img
          alt="Third slide"
          className="d-block w-100"
          src="cinemas.jpg"
        />
      </div>
      <div
        className="carousel-item"
        interval={3000}
      >
        <img
          alt="Third slide"
          className="d-block w-100"
          src="cinema2.jpg"
        />
      </div>
    </div>
  </div>
  <br />
  <h2
    style={
      Object {
        "color": "white",
        "fontWeight": "400",
        "fontfamily": "isonormregular, sans-serif",
        "letterSpacing": ".15em",
        "lineHeight": "1.1",
        "textTransform": "uppercase",
      }
    }
  >
    Book your tickets 
  </h2>
  <br />
  <div
    className="container"
  >
    <div
      className="text-center card"
      style={
        Object {
          "backgroundColor": "#212529",
          "border": "solid",
          "borderColor": "black",
          "color": "white",
        }
      }
    >
      <div
        className="card-header"
      >
        Ticket Select
      </div>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          <u>
            Please fill out the form to book tickets
          </u>
        </div>
        <p
          className="card-text"
        >
          <h5>
            Loading Form
          </h5>
        </p>
      </div>
      <div
        className="text-muted card-footer"
      />
    </div>
  </div>
</div>
`;
